src/code/z_bgcheck.c

```diff
@@ -2952,139 +2945,145 @@ void DynaPoly_ExpandSRT(PlayState* play, DynaCollisionContext* dyna, s32 bgId, s
-            dyna->bgActors[bgId].curTransform.pos.x, dyna->bgActors[bgId].curTransform.pos.y,
-            dyna->bgActors[bgId].curTransform.pos.z);
-
-        if ((pbgdata->numVertices != 0) && (pbgdata->numPolygons != 0)) {
-            s32 j;
-            numVtxInverse = 1.0f / pbgdata->numVertices;
-            newCenterPoint.x = newCenterPoint.y = newCenterPoint.z = 0.0f;
-            for (i = 0; i < pbgdata->numVertices; i++) {
-                Vec3f vtx;  // spF4
-                Vec3f vtxT; // spE8 Vtx after mtx transform
-
-                Math_Vec3s_ToVec3f(&vtx, &pbgdata->vtxList[i]);
-                SkinMatrix_Vec3fMtxFMultXYZ(&D_801EDA40, &vtx, &vtxT);
-                BgCheck_Vec3fToVec3s(&dyna->vtxList[*vtxStartIndex + i], &vtxT);
-
-                if (i == 0) {
-                    dyna->bgActors[bgId].minY = dyna->bgActors[bgId].maxY = vtxT.y;
-                } else if (vtxT.y < dyna->bgActors[bgId].minY) {
-                    dyna->bgActors[bgId].minY = vtxT.y;
-                } else if (dyna->bgActors[bgId].maxY < vtxT.y) {
-                    dyna->bgActors[bgId].maxY = vtxT.y;
-                }
-                newCenterPoint.x += vtxT.x;
-                newCenterPoint.y += vtxT.y;
-                newCenterPoint.z += vtxT.z;
-            }
+        return;
+    }
 
-            newCenterPoint.x *= numVtxInverse;
-            newCenterPoint.y *= numVtxInverse;
-            newCenterPoint.z *= numVtxInverse;
-            sphere->center.x = newCenterPoint.x;
-            sphere->center.y = newCenterPoint.y;
-            sphere->center.z = newCenterPoint.z;
-            newRadiusSq = -100.0f;
-
-            for (i = 0, j = *vtxStartIndex; i < pbgdata->numVertices; i++, j++) {
-                f32 radiusSq;
-
-                newVtx.x = dyna->vtxList[j].x;
-                newVtx.y = dyna->vtxList[j].y;
-                newVtx.z = dyna->vtxList[j].z;
-                radiusSq = Math3D_Vec3fDistSq(&newVtx, &newCenterPoint);
-                if (newRadiusSq < radiusSq) {
-                    newRadiusSq = radiusSq;
-                }
+    SkinMatrix_SetScaleRotateYRPTranslate(
+        &D_801EDA40, dyna->bgActors[bgId].curTransform.scale.x, dyna->bgActors[bgId].curTransform.scale.y,
+        dyna->bgActors[bgId].curTransform.scale.z, dyna->bgActors[bgId].curTransform.rot.x,
+        dyna->bgActors[bgId].curTransform.rot.y, dyna->bgActors[bgId].curTransform.rot.z,
+        dyna->bgActors[bgId].curTransform.pos.x, dyna->bgActors[bgId].curTransform.pos.y,
+        dyna->bgActors[bgId].curTransform.pos.z);
+
+    if ((pbgdata->numVertices != 0) && (pbgdata->numPolygons != 0)) {
+        f32 radiusSq;
+
+        numVtxInverse = 1.0f / pbgdata->numVertices;
+        newCenterPoint.x = newCenterPoint.y = newCenterPoint.z = 0.0f;
+        for (i = 0; i < pbgdata->numVertices; i++) {
+            Vec3f vtx;
+            Vec3f vtxT; // Vtx after mtx transform
```

Comment:
Does it make sense to name this `vtx`?

Reply:
Since it's not of type `Vtx`? Seems like a deep rabbit hole of a lot of `Vec3s`/`Vec3f` data being called vertices here, which makes sense from a bigger point of view as that's what a point on a poly is called, but then is confusing because of the more in-depth `Vtx` struct which contains the position and additional info, while this only contains the position of the vtx.

Reply:
I feel you'd have bigger issues if you tried naming it something else. To me the overlap between Vtx the graphics primitive and vtx to refer to poly vertices is fairly easy to understand when considering problem context.

Reply:
The only reasonable alternative I can think of would be naming it `vtxPos` when it's not the vtx struct but instead the position of the vtx. And when it's local scope like this, maybe even `pos`. but idk